BothamNetworking/BothamAPIClientError.swift

```diff
@@ -8,12 +8,32 @@
 
 import Foundation
 
-public enum BothamAPIClientError: ErrorType {
+public enum BothamAPIClientError: ErrorType, Equatable {
 
     case HTTPResponseError(statusCode: Int, body: NSData)
     case NetworkError
     case HTTPClientError(error: NSError)
     case ParsingError(error: NSError)
     case UnsupportedURLScheme
+    case RetryError
```

Comment:
We should try to keep the naming close to conventions when possible, and avoid to repeat the name Error in the case.

From the documention: 
```Swift
enum VendingMachineError: ErrorType {
    case InvalidSelection
    case InsufficientFunds(coinsNeeded: Int)
    case OutOfStock
}

VendingMachineError.InsufficientFunds(coinsNeeded: 5)
```

In our case:

```Swift
BothamAPIClientError.Retryable
```